sql/catalyst/src/test/scala/org/apache/spark/sql/types/StructTypeSuite.scala

```diff
@@ -73,6 +73,15 @@ class StructTypeSuite extends SparkFunSuite with SQLHelper {
     assert(struct.toDDL == """`b` BOOLEAN COMMENT 'Field\'s comment'""")
   }
 
+  test("SPARK-33846: toDDL should include nested StructType fields' comments") {
+    val nestedStruct = new StructType()
+      .add(StructField("b", StringType).withComment("Nested comment"))
+    val struct = new StructType()
+      .add(StructField("a", nestedStruct).withComment("comment"))
+    val ddl = "`a` STRUCT<`b`: STRING COMMENT 'Nested comment'> COMMENT 'comment'"
+
+    assert(struct.toDDL == ddl)
```

Comment:
Could you add tests to check if `StructType.fromDDL(struct.toDDL)` can work correctly?